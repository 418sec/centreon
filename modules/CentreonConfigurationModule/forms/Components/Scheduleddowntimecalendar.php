<?php
/*
 * Copyright 2005-2015 CENTREON
 * Centreon is developped by : Julien Mathis and Romain Le Merlus under
 * GPL Licence 2.0.
 *
 * This program is free software; you can redistribute it and/or modify it under
 * the terms of the GNU General Public License as published by the Free Software
 * Foundation ; either version 2 of the License.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT ANY
 * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
 * PARTICULAR PURPOSE. See the GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along with
 * this program; if not, see <http://www.gnu.org/licenses>.
 *
 * Linking this program statically or dynamically with other modules is making a
 * combined work based on this program. Thus, the terms and conditions of the GNU
 * General Public License cover the whole combination.
 *
 * As a special exception, the copyright holders of this program give CENTREON
 * permission to link this program with independent modules to produce an executable,
 * regardless of the license terms of these independent modules, and to copy and
 * distribute the resulting executable under terms of CENTREON choice, provided that
 * CENTREON also meet, for each linked independent module, the terms  and conditions
 * of the license of that module. An independent module is a module which is not
 * derived from this program. If you modify this program, you may extend this
 * exception to your version of the program, but you are not obliged to do so. If you
 * do not wish to do so, delete this exception statement from your version.
 *
 * For more information : contact@centreon.com
 *
 */

namespace CentreonConfiguration\Forms\Components;

use Centreon\Internal\Di;
use Centreon\Internal\Form\Component\Component;

/**
 * Component for select the period
 *
 * @author Maximilien Bersoult <mbersoult@centreon.com>
 * @package CentreonConfiguration
 * @subpackage Form
 * @version 3.0.0
 */
class Scheduleddowntimecalendar extends Component
{
    /**
     * Generate and return the html for the element
     *
     * @param array $element The element to parse
     * @return array
     */
    public static function renderHtmlInput(array $element)
    {
        $tpl = Di::getDefault()->get('template');
        $weeklyDays = array(
            0 => "sunday",
            1 => "monday",
            2 => "tuesday",
            3 => "wednesday",
            4 => "thursday",
            5 => "friday",
            6 => "saturday"
        );

        $tpl->assign('element', $element);
        $tpl->assign('weeklyDays', $weeklyDays);
        $tpl->addCss('centreon.scheduled-downtime.css', 'centreon-configuration');
        $tpl->addCss('bootstrap-datetimepicker.min.css');
        $tpl->addJs('centreon.scheduled-downtime.js', 'bottom', 'centreon-configuration');
        $tpl->addJs('hogan-3.0.0.min.js');
        $tpl->addJs('bootstrap-datetimepicker.min.js');
         
        return array(
            'html' => $tpl->fetch('file:[CentreonConfigurationModule]/forms/components/scheduled_downtime_calendar.tpl'),
        );
    }
}
