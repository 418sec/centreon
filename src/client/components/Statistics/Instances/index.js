export { default as Instances } from './InstancesContainer';