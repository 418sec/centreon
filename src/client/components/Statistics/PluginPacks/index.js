export { default as PluginPacks } from './PluginPacksContainer';