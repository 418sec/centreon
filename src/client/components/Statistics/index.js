export { default as Statistics } from './StatisticsContainer';