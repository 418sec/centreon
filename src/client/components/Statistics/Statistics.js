import React from 'react'
import { withStyles } from 'material-ui/styles'
import { HeaderBar } from 'Components/HeaderBar'
import { Route, Switch, Redirect } from 'react-router-dom'

const styles = theme => ({
  title: {
    align: 'center'
  },
  progress: {
    position: 'absolute',
    top: '50%',
    left: '50%',
    zIndex: 1,
  }
})

const Statistics = ({ entries }) => (
  <section>
    <Switch>
      {entries.map((entry, index) => {
        return (
          <Route
            key={'stats' + index}
            path={entry.path}
            exact
            component={entry.component}
          />
        )
      })}
      <Redirect to={entries[0].path} />
    </Switch>
  </section>
)

export default withStyles(styles)(Statistics)