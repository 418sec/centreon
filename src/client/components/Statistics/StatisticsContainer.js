import React, { Component } from 'react'
import { connect } from 'react-redux'
import Statistics from './Statistics'
import { PluginPacks } from 'Components/Statistics/PluginPacks'
import { Instances } from 'Components/Statistics/Instances'
import { cleanHeader, replaceTitle, replaceEntries } from 'Redux/header/headerActions'


class StatisticsContainer extends Component {

  constructor (props) {
    super(props)

    const { match } = props

    this.state = {
      entries: [
        {
          name: 'Plugin Packs',
          path: `${match.path}/pluginpacks`,
          component: PluginPacks
        },
        {
          name: 'Instances',
          path: `${match.path}/instances`,
          component: Instances
        },
      ]
    }
  }

  componentDidMount = () => {
    const { cleanHeader, replaceHeaderTitle, replaceHeaderEntries } = this.props
    const { entries } = this.state

    cleanHeader()
    replaceHeaderTitle('Statistics')
    replaceHeaderEntries(entries)
  }

  render = () => {
    const { entries } = this.state

    return (
      <Statistics
        entries={entries}
      />
    )
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    cleanHeader: () => {
      return dispatch(cleanHeader())
    },
    replaceHeaderTitle: (title) => {
      return dispatch(replaceTitle(title))
    },
    replaceHeaderEntries: (entries) => {
      return dispatch(replaceEntries(entries))
    },
  }
}

export default connect(null, mapDispatchToProps)(StatisticsContainer)