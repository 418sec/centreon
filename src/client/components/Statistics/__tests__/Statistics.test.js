/* eslint-env jest */
import React from 'react'
import Enzyme from 'enzyme'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import { Statistics } from 'Components/Statistics'
import configureStore from 'redux-mock-store'

Enzyme.configure({ adapter: new Adapter() })

describe('Statistics', () => {
  const mockStore = configureStore()
  const store = mockStore({})
  let component

  it('renders correctly', () => {
    component = shallow(
      <Statistics
        match={{path: '/app'}}
        store={store}
      />
    ).dive().dive().dive()
    expect(toJson(component)).toMatchSnapshot()
  })
})