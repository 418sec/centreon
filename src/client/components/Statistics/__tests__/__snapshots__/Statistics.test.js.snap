// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Statistics renders correctly 1`] = `
<section>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      key="stats0"
      path="/app/pluginpacks"
    />
    <Route
      component={[Function]}
      exact={true}
      key="stats1"
      path="/app/instances"
    />
    <Redirect
      push={false}
      to="/app/pluginpacks"
    />
  </Switch>
</section>
`;
